x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -320,12 +320,21 @@ public void testDataStreams_autoFollowAfterDataStreamCreated() throws Exception
 
         final int initialNumDocs = 16;
         int initialNumberOfSuccessfulFollowedIndices = getNumberOfSuccessfulFollowedIndices();
-        final String dataStreamName = "logs-syslog-prod";
-        final String autoFollowPatternName = getTestName().toLowerCase(Locale.ROOT);
+        // The data stream name shouldn't match with builtin ilm policies to avoid test instabilities.
+        // (the manual rollover that happens in this test, may cause ilm to add `index.lifecycle.indexing_complete` setting,
+        // which causes explicit follow index api call to fail in this test)
+        final String dataStreamName = "log-syslog-prod";
+        // Because the builtin logs template isn't used, a template should be defined here.
+        Request putComposableIndexTemplateRequest = new Request("POST", "/_index_template/" + getTestName().toLowerCase(Locale.ROOT));
+        putComposableIndexTemplateRequest.setJsonEntity("{\"index_patterns\": [\"log-syslog-*\"], \"data_stream\": {}}");
```

Comment:
I find such generic patterns very trappy in CCR rest tests, because the cluster is not cleaned up between tests cases. For example, there is a very similar `logs-syslog-prod` data stream in `FollowIndexIT`.

We should maybe use more specific names like `test-auto-follow-backing-indices-*` ?

Reply:
👍 I will update the template to use a specific template.